cdap-app-templates/cdap-etl/cdap-etl-batch/src/main/java/co/cask/cdap/etl/batch/mapreduce/ETLMapReduce.java

```diff
@@ -110,11 +113,29 @@ public void configure() {
     setMapperResources(phaseSpec.getResources());
     setDriverResources(phaseSpec.getResources());
 
+    // These should never happen unless there is a bug in the planner. The planner is supposed to ensure this.
     if (phaseSpec.getPhase().getSources().size() != 1) {
-      throw new IllegalArgumentException("Pipeline must contain exactly one source.");
+      throw new IllegalArgumentException("Pipeline phase must contain exactly one source.");
     }
     if (phaseSpec.getPhase().getSinks().isEmpty()) {
-      throw new IllegalArgumentException("Pipeline must contain at least one sink.");
+      throw new IllegalArgumentException("Pipeline phase must contain at least one sink.");
+    }
+    Set<StageInfo> aggregators = phaseSpec.getPhase().getStagesOfType(BatchAggregator.PLUGIN_TYPE);
+    if (aggregators.size() > 1) {
+      throw new IllegalArgumentException("Pipeline phase cannot have more than one aggregator.");
+    } else if (!aggregators.isEmpty()) {
+      String aggregatorName = aggregators.iterator().next().getName();
+      PipelinePhase mapperPipeline = phaseSpec.getPhase().subsetTo(ImmutableSet.of(aggregatorName));
+      for (StageInfo stageInfo : mapperPipeline) {
+        // error datasets are not supported in the map phase of a mapreduce, because we can only
```

Comment:
perhaps the terminology is confusing here, but a 'stage' is a node in the pipeline and has a unique name, so at least they should be able to identify the problem node.  Why it is a problem will be very confusing though, as it depends on where it sits in the pipeline.... I'm tempted to just remove support altogether